optuna/storages/redis.py

```diff
@@ -0,0 +1,570 @@
+import copy
+from datetime import datetime
+import pickle
+import redis
+
+from optuna import distributions, exceptions  # NOQA
+from optuna.storages import base
+from optuna.storages.base import DEFAULT_STUDY_NAME_PREFIX
+from optuna import structs
+from optuna import type_checking
+
+if type_checking.TYPE_CHECKING:
+    from typing import Any  # NOQA
+    from typing import Dict  # NOQA
+    from typing import List  # NOQA
+    from typing import Optional  # NOQA
+
+
+class RedisStorage(base.BaseStorage):
```

Comment:
Could you wrap this class using the `experimental` decorator? I'd like to merge this PR rather soon although other maintainers/committers haven't yet had the time to take a proper look at the changes in my understanding. I basically think the internals LGTM.

Reply:
Added @experimental("1.2.0") as suggested.